### ISSUE TITLE
Downloaded items in playlists don't show up as downloaded songs within a given album

### ISSUE THREAD

**Issue by StayBlue** [jmshrv/finamp#420]
Sorry if the title of this issue is a bit confusing. I am unsure if this is intentional, a bug, or needs to be added, but just to be sure, I'll explain the problem anyway. Additionally, I would guess that this issue being fixed would be contingent on #213 also being fixed.

After downloading a playlist of songs and entering offline mode, only the downloaded playlist will appear as downloaded, not the individual albums. Let me provide an example. Say you download a playlist containing four songs in an album, and an album contains ten songs. In offline mode, while viewing your albums, you would expect to see the album, and you can play the four songs that you downloaded within that album. Unfortunately, the only way to play those four songs that you have downloaded would be to open up the playlist where you downloaded those four songs until you download the entire album.

I wouldn't say this is a huge issue, but it'd be a nice-to-have.

Edit: Fixed the wrong issue being referenced.

**Comment by StayBlue** [comment]
Referencing #420 (nice) as a result of a [comment](https://github.com/jmshrv/finamp/issues/462#issuecomment-1690410386) that @jmshrv made in said issue.

**Comment by Chaphasilor** [comment]
@StayBlue the download system has been improved as part of the redesign beta. Tracks should now be matched between playlists and albums, and album meta data will also be downloaded for the offline mode for each song in a playlist. You can check it out here: https://github.com/jmshrv/finamp/releases/tag/0.9.2-beta

I'll close this issue for now since it should be resolved.
